diff --git a/tests/utils_tests/test_text.py b/tests/utils_tests/test_text.py
--- a/tests/utils_tests/test_text.py
+++ b/tests/utils_tests/test_text.py
@@ -192,6 +192,13 @@ def test_slugify(self):
             # given - expected - Unicode?
             ('Hello, World!', 'hello-world', False),
             ('spam & eggs', 'spam-eggs', False),
+            (' multiple---dash and  space ', 'multiple-dash-and-space', False),
+            ('\t whitespace-in-value \n', 'whitespace-in-value', False),
+            ('underscore_in-value', 'underscore_in-value', False),
+            ('__strip__underscore-value___', 'strip__underscore-value', False),
+            ('--strip-dash-value---', 'strip-dash-value', False),
+            ('__strip-mixed-value---', 'strip-mixed-value', False),
+            ('_ -strip-mixed-value _-', 'strip-mixed-value', False),
             ('spam & ıçüş', 'spam-ıçüş', True),
             ('foo ıç bar', 'foo-ıç-bar', True),
             ('    foo ıç bar', 'foo-ıç-bar', True),
